stock_delivery_result_import/wizards/stock_delivery_result_import.py

```diff
@@ -72,14 +74,20 @@ def import_stock_delivery_result(self):
                 )
         if not import_log.error_ids:
             for picking, vals in pick_dict.items():
-                carrier_info = vals["carrier_info"]
+                carrier_info = (
+                    picking.carrier_info_id
+                    if picking.carrier_info_id
+                    else vals["carrier_info"]
+                )
                 tracking_refs = list(set(vals["tracking_refs"]))
+                if picking.carrier_tracking_ref:
+                    tracking_refs.append(picking.carrier_tracking_ref)
                 tracking_refs = ", ".join(ref for ref in tracking_refs)
                 picking.write(
```

Comment:
Update `vals` for the available variables only first, and then do `picking.write(vals)`.